warehouse/query-core/src/test/java/datawave/query/jexl/visitors/TreeFlatteningRebuildingVisitorTest.java

```diff
@@ -1,16 +1,50 @@
 package datawave.query.jexl.visitors;
 
 import datawave.query.jexl.JexlASTHelper;
+import datawave.query.language.parser.jexl.LuceneToJexlQueryParser;
 import org.apache.commons.jexl2.parser.JexlNode;
+import org.apache.commons.jexl2.parser.Parser;
+import org.apache.commons.lang.StringUtils;
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.io.StringReader;
+
 public class TreeFlatteningRebuildingVisitorTest {
     
     @Test
-    public void dontFlattenASTDelayedPredicateTest() throws Exception {
+    public void dontFlattenASTDelayedPredicateAndTest() throws Exception {
         String query = "((ASTDelayedPredicate = true) && (GEO == '1f36c71c71c71c71c7' && (WKT_BYTE_LENGTH >= '+AE0' && WKT_BYTE_LENGTH < '+bE8'))) && GEO >= '1f36c71c71c71c71c7\uDBFF\uDFFF+AE0' && GEO < '1f36c71c71c71c71c8\uDBFF\uDFFF+bE8'";
         JexlNode node = TreeFlatteningRebuildingVisitor.flatten(JexlASTHelper.parseJexlQuery(query));
         Assert.assertEquals(query, JexlStringBuildingVisitor.buildQuery(node));
     }
+    
+    @Test
+    public void dontFlattenASTDelayedPredicateOrTest() throws Exception {
```

Comment:
Should this be 'flattenASTDelayedPredicateOrTest' since we ARE expecting the query to be flattened in this case?

Reply:
changed